app/src/main/AndroidManifest.xml

```diff
@@ -20,10 +20,9 @@
         <activity
             android:name=".SecondActivity"
             android:label="@string/title_activity_second_activity">
-            <intent-filter>
-                <action android:name="android.intent.action.MAIN" />
-                <category android:name="android.intent.category.LAUNCHER" />
-            </intent-filter>
+<!--            <intent-filter>-->
+<!--                <action android:name="android.intent.action.MAIN" />-->
+<!--            </intent-filter>-->
```

Comment:
The category launcher is what was causing the issue with multiple entries for the app showing in the app list.

I commented out the intent filter as well because it does not seem to be necessary for switching between activities within the same application.